lib/reqTemplate.js

```diff
@@ -42,28 +42,36 @@ function _setAtPath(path, spec) {
     return new Function('obj', 'value', code);
 }
 
-/**
- * Validates and prepares the template string - resolves local references
- * and wraps property access to ["..."]
- *
- * @param {string} template an original template string
- * @param {string} defaultLookupPath a default path withing the context which
- *                 to look if template is local
- * @returns {string} validated and
- */
-function prepareTemplateString(template, defaultLookupPath) {
-    if (!/^(\$\.)?([a-zA-Z][a-zA-Z0-9-_]*)(\.[a-zA-Z][a-zA-Z0-9-_]*)*$/.test(template)) {
-        throw new Error('Invalid template ' + template);
+function splitAndPrepareTAsseblyTemplate(templateSpec) {
```

Comment:
Touché. On the upside, it will figure out the brackets are unbalanced. Albeit, adding a matching bracket inside the string will pass unnoticed:

```js
foo { '{}' }
```